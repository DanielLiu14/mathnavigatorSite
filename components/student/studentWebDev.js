'use strict';
require('./studentWebDev.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import {
  description,
  sectionIg,
  sectionTesla,
  sectionProfile,
  sectionYelp
} from './webdev.js';
import { debounce } from 'lodash';
const classnames = require('classnames');

export class StudentWebDevPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: description,
      webdevUrl: "/class/web_design_summer_a",
      showPropzIg: false,
      showPropzTesla: false,
      showPropzProfile1: false,
      showPropzProfile2: false,
      showPropzYelp: false
    }
    this.handleScroll = this.handleScroll.bind(this);
  }

  componentDidMount() {
    this.unbindTimeout = setTimeout(function() {
      this.setState({ showPropzIg: true });
    }.bind(this), 1500);

    window.addEventListener('scroll', debounce(this.handleScroll, 200));

    if (process.env.NODE_ENV === 'production') {
      mixpanel.track("student-webdev");
    }
  }

  componentWillUnmount() {
    if (this.unbindTimeout) {
      clearTimeout(this.unbindTimeout);
    }
    window.removeEventListener('scroll', debounce(this.handleScroll, 200));
  }

  handleScroll(event) {
    var statePropzTesla = this.state.showPropzTesla;
    var statePropzProfile1 = this.state.showPropzProfile1;
    var statePropzProfile2 = this.state.showPropzProfile2;
    var statePropzYelp = this.state.showPropzYelp;

    if (statePropzTesla &&
        statePropzProfile1 &&
        statePropzProfile2 &&
        statePropzYelp) {
      return;
    }

    var showPropzTesla = statePropzTesla;
    var showPropzProfile1 = statePropzProfile1;
    var showPropzProfile2 = statePropzProfile2;
    var showPropzYelp = statePropzYelp;

    var scrollY = event.target.scrollingElement.scrollTop;
    if (!showPropzTesla && scrollY > 320) {
      showPropzTesla = true;
    }
    if (!showPropzProfile1 && scrollY > 600) {
      showPropzProfile1 = true;
    }
    if (!showPropzProfile2 && scrollY > 1100) {
      showPropzProfile2 = true;
    }
    if (!showPropzYelp && scrollY > 1500) {
      showPropzYelp = true;
    }

    var needToSet = statePropzTesla ^ showPropzTesla;
    needToSet = needToSet || (statePropzProfile1 ^ showPropzProfile1);
    needToSet = needToSet || (statePropzProfile2 ^ showPropzProfile2);
    needToSet = needToSet || (statePropzYelp ^ showPropzYelp);

    if (needToSet) {
      this.setState({
        showPropzTesla: showPropzTesla,
        showPropzProfile1: showPropzProfile1,
        showPropzProfile2: showPropzProfile2,
        showPropzYelp: showPropzYelp,
      });
    }
  }

	render() {
		return (
      <div id="view-students-wb">
        <div id="view-students-wb-container">
          <h1>
            <Link to="/programs">Programs</Link> >
            <Link to={this.state.webdevUrl}> Web Development</Link> >
            Overview
          </h1>
          <p> {this.state.description} </p>

          <SectionIG showPropz={this.state.showPropzIg}/>
          <SectionTesla showPropz={this.state.showPropzTesla}/>
          <SectionProfile showPropz1={this.state.showPropzProfile1}
            showPropz2={this.state.showPropzProfile2}/>
          <SectionYelp showPropz={this.state.showPropzYelp}/>
        </div>
      </div>
		);
	}
}

class SectionIG extends React.Component {
  render() {
    const info = sectionIg;
    var propzInfo = generatePropz(info);
    var propzStyle = {
      top: "10px",
      left: "45%"
    };

    return (
      <div className="section ig-header">
        <h2>{info.title}</h2>
        <div className="propz-container">
          <img src={info.imgSrc}/>
          <Propz info={propzInfo} style={propzStyle} direction="up"
            show={this.props.showPropz}/>
        </div>
        <div className="credit">Created by {info.student1}, {info.student2}</div>
        <p>{info.description}</p>
      </div>
    );
  }
}

class SectionTesla extends React.Component {
  render() {
    const info = sectionTesla;
    var propzInfo = generatePropz(info);
    var propzStyle = {
      top: "156px",
      left: "132%"
    };
    return (
      <div className="section tesla">
        <h2>{info.title}</h2>
        <div className="tesla-container">
          <div className="propz-container">
            <img src={info.imgSrc}/>
            <Propz info={propzInfo} style={propzStyle} direction="right"
              show={this.props.showPropz}/>
          </div>
          <div className="social-links">
            <a href="https://www.facebook.com/TeslaMoto" target="_blank">
              <div className="icon social-fb"/>
            </a>
            <a href="https://www.instagram.com/teslamotors/" target="_blank">
              <div className="icon social-ig"/>
            </a>
            <a href="https://twitter.com/tesla" target="_blank">
              <div className="icon social-tw"/>
            </a>
            <a href="https://www.youtube.com/channel/UC5WjFrtBdufl6CZojX3D8dQ" target="_blank">
              <div className="icon social-yt"/>
            </a>
          </div>
        </div>
        <div className="credit">Created by {info.student1}, {info.student2}</div>
        <p>{info.description}</p>
      </div>
    );
  }
}

class SectionProfile extends React.Component {
  render() {
    const info = sectionProfile;
    var propzInfo1 = generatePropz(info.info1);
    var propzStyle1 = {
      top: "0px",
      left: "70%"
    };
    var propzInfo2 = generatePropz(info.info2);
    var propzStyle2 = {
      top: "90%",
      left: "30%"
    };
    return (
      <div className="section profiles">
        <h2>{info.title}</h2>
        <div className="propz-container">
          <img className="img1" src={info.info1.imgSrc}/>
          <Propz info={propzInfo1} style={propzStyle1} direction="up"
            show={this.props.showPropz1}/>
          <div className="credit">Created by {info.info1.student1}, {info.info1.student2}</div>
        </div>
        <div>
          <div className="descriptions">
            <p>{info.info1.description}</p>
            <p>{info.info2.description}</p>
          </div>
          <div className="propz-container">
            <img className="img2" src={info.info2.imgSrc}/>
            <Propz info={propzInfo2} style={propzStyle2} direction="down"
              show={this.props.showPropz2}/>
            <div className="credit">Created by {info.info2.student1}, {info.info2.student2}</div>
          </div>
        </div>
      </div>
    );
  }
}

class SectionYelp extends React.Component {
  render() {
    const info = sectionYelp;
    var propzInfo = generatePropz(info);
    var propzStyle = {
      top: "164px",
      left: "80%"
    };
    return (
      <div className="section yelp">
        <h2>{info.title}</h2>
        <p>{info.description}</p>
        <div className="propz-container">
          <img src={info.imgSrc}/>
          <Propz info={propzInfo} style={propzStyle} direction="right"
            show={this.props.showPropz}/>
        </div>
        <div className="credit">Created by {info.student1}, {info.student2}</div>
      </div>
    );
  }
}

class Propz extends React.Component {
  render() {
    const info = this.props.info;
    const style = this.props.style;
    var propzClasses = classnames("propz", this.props.direction, {
      "show": this.props.show
    });

    return (
      <div className={propzClasses} style={style}>
        <div className="tri"/>
        <div>Created by {info.line1}</div>
        <div>{info.line2}</div>
      </div>
    );
  }
}

function generatePropz(info) {
  return {
    line1: info.student1,
    line2: info.student2
  };
}
