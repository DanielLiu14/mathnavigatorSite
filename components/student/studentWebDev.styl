@import "../static.styl"

#view-students-wb
  width: containerWidthPct
  max-width: containerWidthPx1
  margin: spacing12x auto spacing16x

  & h1
    line-height: lineHeight2
    font-family: fontFamilyRockwell
    font-size: fontSize2
    color: colorPrimary

  & h2
    line-height: lineHeight2
    font-family: fontFamilyRoboto
    font-size: fontSize1

  & a
    color: colorPrimary

  & p
    margin: spacing4x 0 spacing12x
    max-width: propUnit20x
    line-height: lineHeight0
    font-size: fontSize0

#view-students-wb .section
  margin-bottom: spacing12x

  & h2
    font-family: fontFamilyRoboto
    line-height: lineHeight1
    font-size: fontSize1
    color: colorPrimary

    @media screen and (max-width: mediaWidth2)
      text-align: center

  & p
    margin: 0
    max-width: propUnit16x
    color: colorBlackDark

    @media screen and (max-width: mediaWidth2)
      text-align: center

  & .credit
    display: none
    margin-top: spacing1x
    font-size: 12px
    text-align: center

    @media screen and (max-width: mediaWidth2)
      display: block

  & img
    max-width: 100%

  &.ig-header
    & img
      width: 100%
      margin-top: spacing4x

    & p
      margin-top: spacing4x

  &.tesla
    & h2
      margin-bottom: spacing4x

    & .tesla-container
      display: inline-block
      margin-right: spacing8x
      vertical-align: top

      @media screen and (max-width: mediaWidth2)
        display: block
        width: 180px
        margin: 0 auto

    & img
      width: 160px

    & .social-links
      display: block
      margin-top: spacing2x

    & .icon
      display: inline-block
      width: 28px
      height: 28px
      margin: 0 spacing2x
      background-size: cover
      background-position: center
      background-repeat: no-repeat
      cursor: pointer

      &:first-child
        margin-left: 0

      &.social-fb
        background-image: url("../../assets/students/fb_black.png")
        &:hover
          background-image: url("../../assets/students/fb_blue.png")
      &.social-ig
        background-image: url("../../assets/students/ig_black.png")
        &:hover
          background-image: url("../../assets/students/ig_blue.png")
      &.social-tw
        background-image: url("../../assets/students/tw_black.png")
        &:hover
          background-image: url("../../assets/students/tw_blue.png")
      &.social-yt
        width: 28px
        height: 22px
        margin-top: 4px
        vertical-align: top
        background-image: url("../../assets/students/yt_black.png")
        &:hover
          background-image: url("../../assets/students/yt_red.png")

    & p
      display: inline-block
      width: 'calc(100% - 176px - %s)' % spacing8x

      @media screen and (max-width: mediaWidth2)
        display: block
        margin: spacing2x auto 0
        width: 100%

  &.profiles
    & .img1
      height: 240px
      margin: spacing4x 0

      @media screen and (max-width: mediaWidth2)
        display: block
        width: 100%
        height: initial
        margin: spacing4x auto 0

    & .img2
      height: 360px

    & .descriptions
      display: inline-block
      margin-bottom: spacing4x
      width: 'calc(100% - 364px - %s)' % spacing4x
      vertical-align: top

      @media screen and (max-width: mediaWidth2)
        width: 100%
        margin-top: spacing4x
        margin-bottom: spacing2x

      & p
        margin-bottom: spacing4x
        @media screen and (max-width: mediaWidth2)
          max-width: none

      & + .propz-container
        display: inline-block
        width: 280px
        margin-left: spacing4x

        @media screen and (max-width: mediaWidth2)
          display: block
          margin: 0 auto

        & img
          height: auto

  &.yelp
    & p
      margin: spacing4x 0

#view-students-wb .propz-container
  display: inline-block
  position: relative

  & .propz
    position: absolute
    line-height: 20px
    width: 142px
    height: 40px
    padding: spacing1x spacing2x
    border-radius: 8px
    background-color: colorOrange
    color: colorWhite
    font-family: fontFamilyRoboto
    font-size: 14px
    opacity: 0.0
    visibility: hidden
    transition: 0.2s linear all

    &.up
      transform: translate(0, 10%)
    &.down
      transform: translate(0, -10%)
    &.right
      transform: translate(-10%, 0)
    &.left
      transform: translate(10%, 0)

    &.show
      transform: translate(0, 0)
      opacity: 1.0
      visibility: visible

      @media screen and (max-width: mediaWidth2)
        width: 0px
        overflow: hidden
        opacity: 0.0
        visibility: hidden

    & .tri
      position: absolute
      width: 0
      height: 0

    &.up .tri
      top: 100%
      left: 24px
      border-left: 6px solid transparent
      border-right: 6px solid transparent
      border-top: 8px solid colorOrange

    &.down .tri
      bottom: 100%
      right: 24px
      border-left: 6px solid transparent
      border-right: 6px solid transparent
      border-bottom: 8px solid colorOrange

    &.right .tri
      bottom: 22px
      right: 100%
      border-top: 6px solid transparent;
      border-bottom: 6px solid transparent;
      border-right: 8px solid colorOrange;

    &.left .tri
      bottom: 22px
      left: 100%
      border-top: 6px solid transparent;
      border-bottom: 6px solid transparent;
      border-left: 8px solid colorOrange;
