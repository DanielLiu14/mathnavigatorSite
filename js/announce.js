'use strict';
require('./../styl/announce.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { getList } from './repos/announceRepo.js';

export class AnnouncePage extends React.Component {
	render() {
		const announcements = getList();

		const cards = announcements.map((obj, index) =>
			<AnnounceCard key={index} announcement={obj} />
		);

		return (
      <div id="view-announce">
        <div id="view-announce-container">
          <h1>Announcements</h1>
					{cards}
        </div>
      </div>
		);
	}
}

class AnnounceCard extends React.Component {
	render() {
		const announcement = this.props.announcement;
		return (
			<div className="announce-card">
				<h3>{announcement.date}</h3>
				<p>{announcement.message}</p>
				<div className="card-author">~ {announcement.author}</div>
				<div></div>
			</div>
		);
	}
}
