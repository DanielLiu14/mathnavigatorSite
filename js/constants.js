'use strict';
var _ = require('lodash/core');

export const NavLinks = [
  { id: "home", name: "Home", url: "/" },
  { id: "announce", name: "Announcements", url: "/announcements" },
  { id: "programs", name: "Programs", url: "/programs" },
  { id: "contact", name: "Contact", url: "/contact" }
];

export function getNav(id) {
  return _.find(NavLinks, {id: id});
}

export function isPathAt(url) {
  var path = location.hash;
  if (url == getNav("home").url) {
    return path == '#/';
  } else {
    return path.indexOf(url) >= 0;
  }
}
