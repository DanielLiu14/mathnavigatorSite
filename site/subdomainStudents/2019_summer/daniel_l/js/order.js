console.log("My Javascript works");

function init() {
  console.log("jQuery ready!");
  var flavor = window.localStorage.getItem("flavor");
  $("select.milk-tea").val(flavor);
  var ice = window.localStorage.getItem("ice");
  $("select.ice-amount").val(ice);
  var sugar = window.localStorage.getItem("sugar");
  $("select.sugar-amount").val(sugar);  
  var bubbles = window.localStorage.getItem("bubbles");
  var isChecked = bubbles =="true";
  $("input.bubbles").prop("checked", isChecked);
  var grassJelly = window.localStorage.getItem("grass-jelly");
  var isChecked2 = grassJelly =="true";
  $("input.grass-jelly").prop("checked", isChecked);
  var pudding = window.localStorage.getItem("pudding");
  var isChecked3 = pudding =="true";
  $("input.pudding").prop("checked", isChecked);

  // Setup on-click handler
  $("button").on("click", handleClick);
  $(".milk-tea").on("change", handleSelectChange);
  $(".ice-amount").on("change", handleSelectChange);
  $(".sugar-amount").on("change", handleSelectChange);
  $(".bubbles").on("change", handleCheckboxChange);
  $(".grass-jelly").on("change", handleCheckboxChange);
  $(".pudding").on("change", handleCheckboxChange);
}


function handleSelectChange(e) {
  var targetVal = e.target.value;
  var targetClass = e.target.className;
  if (targetClass == "milk-tea"){
	window.localStorage.setItem("flavor", targetVal);
  }
  if (targetClass == "ice-amount"){
	window.localStorage.setItem("ice", targetVal);
  }
  if (targetClass == "sugar-amount"){
	window.localStorage.setItem("sugar", targetVal);
  }
}

function handleCheckboxChange(e) {
  var targetVal = e.target.checked;
  var targetClass = e.target.className;
  if (targetClass == "bubbles"){
    window.localStorage.setItem("bubbles", targetVal);
  }
  if (targetClass == "grass-jelly"){
    window.localStorage.setItem("grass-jelly", targetVal);
  }
  if (targetClass == "pudding"){
    window.localStorage.setItem("pudding", targetVal);
  }
}

function handleClick() {
  var milkTeaSelected = $(".milk-tea").val();
  var iceAmount = $(".ice-amount").val();
  var sugarAmount = $(".sugar-amount").val();
  var name = $(".enterName").val();
  var url = "html/receipt.html?flavor="+milkTeaSelected+"&ice="+iceAmount+"&sugar="+sugarAmount+"&name="+name;
  if ($(".bubbles").prop("checked") == true) {
	  url += "&toppings=bubbles";
  }
  if ($(".grass-jelly").prop("checked") == true) {
	  url += "&toppings=grassjelly";
  }
  if ($(".pudding").prop("checked") == true) {
	  url += "&toppings=pudding";
  }
  console.log(url);
  window.location = url;
}


$(document).ready(init);