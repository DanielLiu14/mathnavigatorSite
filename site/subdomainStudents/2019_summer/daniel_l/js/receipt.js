console.log("My Javascript works");

var toppings = [];

var imgMap = {
  brownsugar: "../assets/caramel.jpg",
  jasmine: "../assets/cappuccino.jpg",
  oolong: "../assets/black_coffee.jpg",
  thai: "../assets/latte.jpg",
  taro: "../assets/cold_brew.jpg"
};

var flavorStringMap = {
  brownsugar: "Caramel Coffee",
  jasmine: "Cappuccino",
  oolong: "Black Coffee",
  thai: "Latte",
  taro: "Cold Brew"
};

var iceStringMap = {
  full: "More ice",
  half: "Normal ice",
  none: "No Ice"
};

var sugarStringMap = {
  hundred: "Two packets of sugar",
  fifty: "One packet of sugar",
  twentyfive: "No sugar"
};

var toppingStringMap = {
  bubbles: "Foam",
  grassjelly: "Expresso",
  pudding: "Cream"
}

var priceMap = {
  brownsugar: 4,
  jasmine: 2.5,
  oolong: 2.5,
  thai: 3,
  taro: 3
};

function init() {
  console.log("jQuery ready!");
  var map = convertSearch(window.location.search);
  var name1 = map["name"];
  var name2 = name1.replace("%20"," ");
  var totalPrice = priceMap[map["flavor"]]+.5*toppings.length;
  console.log(JSON.stringify(map));
  console.log(JSON.stringify(toppings));
  $("img").attr("src",imgMap[map["flavor"]]);
  $(".greetings").text("Hi " + name2 + ", thanks for ordering at Starbucks! This is your receipt:");
  $(".flavor").text(flavorStringMap[map["flavor"]]);
  $(".ice").text("Ice Amount: " + iceStringMap[map["ice"]]);
  $(".sugar").text("Sugar Amount: " + sugarStringMap[map["sugar"]]);
  $(".price").text("Price: " + totalPrice + " dollars");
  if (toppings.length == 0){
    $(".toppings").text("Toppings: No Toppings");
  }
  if (toppings.length == 1){
	$(".toppings").text("Toppings: "+toppingStringMap[toppings[0]]);
  }
  if (toppings.length == 2){
	$(".toppings").text("Toppings: "+toppingStringMap[toppings[0]]+", "+toppingStringMap[toppings[1]]);
  }
  if (toppings.length == 3){
	$(".toppings").text("Toppings: "+toppingStringMap[toppings[0]]+", "+toppingStringMap[toppings[1]]+", "+toppingStringMap[toppings[2]]);
  }
}

$(document).ready(init);

function convertSearch(search) {
  var map = {};
  search = search || "";
  search = search.substring(1);
  var pairList = search.split("&");
  for (var i=0; i < pairList.length; i++) {
    var pair = pairList[i];
    var split = pair.split("=");
    var key = split[0];
    var value = split[1];
	if(key == "toppings") {
		toppings.push(value);
	} else {
		map[key] = value;
	}
  }
  return map;
}