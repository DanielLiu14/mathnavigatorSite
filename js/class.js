'use strict';
require('./../styl/class.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import { ErrorPage } from './error.js';
const classNames = require('classnames');
import {
  getLocation,
  getPrereqs,
  getProgramClass,
  getProgramByIds,
  getSessions
} from './repos/mainRepo.js';

function generatePrereqs(prereqIds) {
  var prereqsPrograms = getProgramByIds(prereqIds);
  var prereqsNames = prereqsPrograms.map(function(program) {
    return program ? program.title : null;
  });
  return prereqsNames.join(", ");
}

function generateLocation(locationObj) {
  return (
    <div className="class-lines">
      {locationObj.name}<br/>
      {locationObj.address1}<br/>
      {locationObj.address2}<br/>
      {locationObj.address3}
    </div>
  );
}

function generateTimes(classObj) {
  const times = classObj.times.map((time, index) =>
    <div key={index}>{time}</div>
  );
  return (
    <div className="class-lines">
      {times}
      <div>
        First session: {classObj.startDate}<br/>
        Last session: {classObj.endDate}
      </div>
    </div>
  );
}

function generatePricing(pricePerSession, numSessions) {
  let line1;
  let line2;
  let line3;
  if (numSessions > 0) {
    line1 = <div>{"Number of sessions: " + numSessions}</div>;
    line3 = <div>{"Total: $" + numSessions * pricePerSession}</div>;
  } else {

  }
  line2 = <div>{"Price per session: $" + pricePerSession}</div>;
  return (
    <div className="class-lines">
      {line1}
      {line2}
      {line3}
    </div>
  );
}

function generateSchedules(sessions) {
  if (!sessions || sessions.length == 0) {
    return (
      <div className="class-lines not-avail">
        No schedule available at the moment. <br/>
        Please check again later.
      </div>
    );
  }

  var sessionIndex = 0;
  const sessionLines = sessions.map(function(session, index) {
    var text1 = "";
    var text2 = ""
    if (session.canceled) {
      text1 = "Canceled";
      text2 = "";
    } else {
      text1 = session.time;
      text2 = session.notes;
      sessionIndex++;
    }
    return (
      <SessionLine
        key = {session.key + index}
        sessionIndex = {sessionIndex}
        date = {session.date}
        canceled = {session.canceled}
        text1 = {text1}
        text2 = {text2}
      />
    );
  });
  return (
    <ul>
      {sessionLines}
    </ul>
  );
}

export class ClassPage extends React.Component {
  render () {
    var valid = true;
    const key = this.props.slug;
    const pair = getProgramClass(key);
    valid = valid && Boolean(pair);
    valid = valid && Boolean(pair.programObj);
    valid = valid && Boolean(pair.classObj);

    var content;
    if (valid) {
      content = <ClassContent classKey={key} pair={pair}/>;
    } else {
      content = <ErrorPage classDNE={key}/>;
    }

    return (
      <div> {content} </div>
    );
  }
}

class ClassContent extends React.Component {
	render() {
    const classKey = this.props.classKey;
    const pair = this.props.pair;

    // Variables
    const programObj = pair.programObj;
    const classObj = pair.classObj;
    const locationObj = getLocation(classObj.locationId);
    let sessions = getSessions(classKey);
    sessions = sessions ? sessions : [];
    var sessionCounter = 0;
    sessions.forEach(function(session) {
      if (!session.canceled) {
        sessionCounter++;
      }
    });
    var programPrereqs = getPrereqs(programObj.programId);
    const prereqIds = programPrereqs ? programPrereqs.requiredProgramIds : [];

    // Components
    const classFullName = programObj.title + " " + classObj.className;
    const prereqs = generatePrereqs(prereqIds);
    let prereqsLine;
    if (prereqs) {
      prereqsLine = <div>{"Prequirements: " + prereqs}<br/></div>;
    }
    const textLocation = generateLocation(locationObj);
    const textTimes = generateTimes(classObj);
    const textPricing = generatePricing(classObj.pricePerSession, sessionCounter);
    const schedules = generateSchedules(sessions);

		return (
      <div id="view-class">
        <div id="view-class-container">
          <h1>
            <Link to="/programs">Programs</Link> > {classFullName}
          </h1>

          <div id="class-info-container">
            <div className="class-info-1">
              Grades: {programObj.grade1} - {programObj.grade2}<br/>
              {prereqsLine}
              {programObj.description}
            </div>

            <div className="class-info-2">
              <b>Location</b>
              {textLocation}
              <b>Times</b>
              {textTimes}
              <b>Pricing</b>
              {textPricing}
              <Link to={"/contact"}>
                <button className="inverted">Register</button>
              </Link>
            </div>
          </div>

          <div id="view-schedule">
            <b>Schedule</b>
            {schedules}
          </div>

          <div id="view-questions">
            Questions? <Link to="/contact">Contact Us</Link>
          </div>
          <Link to="/programs">
            <button className="inverted">&#60; More Programs</button>
          </Link>
        </div>
      </div>
		);
	}
}

class SessionLine extends React.Component {
  render() {
    var sessionIndex = this.props.sessionIndex;
    var date = this.props.date;
    var canceled = this.props.canceled;
    var text1 = this.props.text1;
    var text2 = this.props.text2;

    const classText1 = classNames("", {
      alert: canceled
    });
    return (
      <li>
        <div className="line-left">
          <div className="line-index">{canceled ? "" : sessionIndex}</div>
          <div className="line-date">{date}</div>
        </div>
        <div className="line-right">
          <div className={classText1}>{text1}</div>
          <div className="class-line alert">{text2}</div>
        </div>
      </li>
    );
  }
}
