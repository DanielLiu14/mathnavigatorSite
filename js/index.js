'use strict';
require('../styl/app.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import {
  HashRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import { Header } from './header.js';
import { HomePage } from './home.js';
import { AnnouncePage } from './announce.js';
import { ProgramsPage } from './programs.js';
import { ContactPage } from './contact.js';
import { Footer } from './footer.js';
import { ClassPage } from './class.js';
import { ErrorPage } from './error.js';

const Home = () => <HomePage/>;
const Announce = () => <AnnouncePage/>;
const Programs = () => <ProgramsPage/>;
const Contact = () => <ContactPage/>;
const ClassPageWithSlug = ({match}) => <ClassPage slug={match.params.slug}/>;
const Error = () => <ErrorPage/>;

class MainContainer extends React.Component {
	render() {
		return (
      <Router onUpdate={() => window.scrollTo(0, 0)}>
        <div>
          <Header/>
          <Switch>
            <Route path="/" exact component={Home}/>
            <Route path="/announcements" component={Announce}/>
            <Route path="/programs" component={Programs}/>
            <Route path="/contact" component={Contact}/>
            <Route path="/class/:slug" component={ClassPageWithSlug}/>
            <Route path="/" component={Error}/>
          </Switch>
          <Footer/>
        </div>
      </Router>
		);
	}
}

ReactDOM.render(
  <MainContainer/>,
  document.getElementById('root')
);
