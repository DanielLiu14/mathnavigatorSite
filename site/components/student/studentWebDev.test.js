import React from "react";
import Enzyme, { shallow } from "enzyme";
import { StudentWebDevPage } from "./studentWebDev.js";
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe("test", () => {
  const component = shallow(<StudentWebDevPage/>);

  test("renders", () => {
    expect(component.exists()).toBe(true);
    expect(component.find("h2").text()).toBe("Website Design & Development");
  });

  test("snapshot", () => {
    const tree = renderer.create(
      <Router>
        <StudentWebDevPage/>
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
