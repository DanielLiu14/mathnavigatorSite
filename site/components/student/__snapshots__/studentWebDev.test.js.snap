// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  id="view-students-wb"
>
  <div
    id="view-students-wb-container"
  >
    <h1>
      Overview for Students
    </h1>
    <h2>
      Website Design & Development
    </h2>
    <p>
       
      When enrolled in our Website Development Program, students will create multiple mini-projects as exercises to apply their knowledge in website development. These small projects will allow students to recreate popular website designs and to creatively design their own!
       
    </p>
    <div
      className="section ig-header"
    >
      <h2>
        Instagram Header
      </h2>
      <div
        className="propz-container"
      >
        <img
          src="test-file-stub"
        />
        <div
          className="propz up"
          style={
            Object {
              "left": "45%",
              "top": "10px",
            }
          }
        >
          <div
            className="tri"
          />
          <div>
            Created by 
            Evelyn
          </div>
          <div>
            8th Grade
          </div>
        </div>
      </div>
      <div
        className="credit"
      >
        Created by 
        Evelyn
        , 
        8th Grade
      </div>
      <p>
        Our first project in this class is recreating the header of Instagram's website. We teach students how to render images and position them to perfectly match the real website's header pixel by pixel!
      </p>
    </div>
    <div
      className="section tesla"
    >
      <h2>
        Tesla's Social Media
      </h2>
      <div
        className="tesla-container"
      >
        <div
          className="propz-container"
        >
          <img
            src="test-file-stub"
          />
          <div
            className="propz right"
            style={
              Object {
                "left": "132%",
                "top": "156px",
              }
            }
          >
            <div
              className="tri"
            />
            <div>
              Created by 
              Richard
            </div>
            <div>
              12th Grade
            </div>
          </div>
        </div>
        <div
          className="social-links"
        >
          <a
            href="https://www.facebook.com/TeslaMoto"
            target="_blank"
          >
            <div
              className="icon social-fb"
            />
          </a>
          <a
            href="https://www.instagram.com/teslamotors/"
            target="_blank"
          >
            <div
              className="icon social-ig"
            />
          </a>
          <a
            href="https://twitter.com/tesla"
            target="_blank"
          >
            <div
              className="icon social-tw"
            />
          </a>
          <a
            href="https://www.youtube.com/channel/UC5WjFrtBdufl6CZojX3D8dQ"
            target="_blank"
          >
            <div
              className="icon social-yt"
            />
          </a>
        </div>
      </div>
      <div
        className="credit"
      >
        Created by 
        Richard
        , 
        12th Grade
      </div>
      <p>
        Many large companies provide links to their social media pages in order to increase user engagement. But simply providing hyperlinks is boring - interactive icons are better. Hover over the social media icons under the Tesla logo to try it yourself!
      </p>
    </div>
    <div
      className="section profiles"
    >
      <h2>
        Professional Profile Cards
      </h2>
      <div
        className="propz-container"
      >
        <img
          className="img1"
          src="test-file-stub"
        />
        <div
          className="propz up"
          style={
            Object {
              "left": "70%",
              "top": "0px",
            }
          }
        >
          <div
            className="tri"
          />
          <div>
            Created by 
            Gaoxing
          </div>
          <div>
            12th Grade
          </div>
        </div>
        <div
          className="credit"
        >
          Created by 
          Gaoxing
          , 
          12th Grade
        </div>
      </div>
      <div>
        <div
          className="descriptions"
        >
          <p>
            Web design is a crucial element for modern websites to succeed. We encourage students to make their websites not only functional, but also aesthetically pleasing. In this project, we study how to create design-forward cards to present people or creative content in a professional fashion.
          </p>
          <p>
            With the introduction of mobile phones and tablets, websites are no longer browsed only with desktops. So we have our students create responsive websites that alter their structure based on browser screen size. Make your browser window larger or smaller to see the difference!
          </p>
        </div>
        <div
          className="propz-container"
        >
          <img
            className="img2"
            src="test-file-stub"
          />
          <div
            className="propz down"
            style={
              Object {
                "left": "30%",
                "top": "90%",
              }
            }
          >
            <div
              className="tri"
            />
            <div>
              Created by 
              Jessica
            </div>
            <div>
              12th Grade
            </div>
          </div>
          <div
            className="credit"
          >
            Created by 
            Jessica
            , 
            12th Grade
          </div>
        </div>
      </div>
    </div>
    <div
      className="section yelp"
    >
      <h2>
        Yelp Search Results
      </h2>
      <p>
        Using some advanced Javascript techniques, you can produce long lists of content like Yelp's search results. In our class, students simulate the same functionality to produce pixel-perfect Yelp content!
      </p>
      <div
        className="propz-container"
      >
        <img
          src="test-file-stub"
        />
        <div
          className="propz right"
          style={
            Object {
              "left": "80%",
              "top": "164px",
            }
          }
        >
          <div
            className="tri"
          />
          <div>
            Created by 
            Daniel
          </div>
          <div>
            11th Grade
          </div>
        </div>
      </div>
      <div
        className="credit"
      >
        Created by 
        Daniel
        , 
        11th Grade
      </div>
    </div>
    <h3>
      And more coming soon!
    </h3>
  </div>
</div>
`;
